src/main/java/org/openstreetmap/atlas/utilities/caching/ConcurrentResourceCache.java

```diff
@@ -34,31 +39,14 @@
      *            the default fetcher
      */
     public ConcurrentResourceCache(final CachingStrategy cachingStrategy,
-            final Function<URI, Resource> fetcher)
+            final Function<URI, Optional<Resource>> fetcher)
     {
         this.cachingStrategy = cachingStrategy;
         this.fetcher = fetcher;
+        this.cacheID = UUID.randomUUID();
```

Comment:
If this is used just for logging it might be helpful to have the name of the Cache with a unique ID. 
Something to the effective of "ConcurrentResourceCache instance {}: <message here>". Theoretically you should know what caching you're using but in the rare case of using two different types of caching in a job, it would be hard to differentiate without the cache name.

Reply:
So the logging framework will already show you the class and the line of code from which each message originates, eg.
```
2018-10-29 07:43:21 WARN  [main] ConcurrentResourceCache:63 - CacheID e18be71c-21f5-46fa-99cc-ca70bdf9fded: cache fetch failed, falling back to default fetcher...
```
That being said, it won't show the actual subclass implementation, should the log be triggered by a class that extended `ConcurrentResourceCache`. That could be remedied by storing `this.getClass().getName()` as a member variable, but it would also add more noise to the logs. Thoughts?

Reply:
I'll put in my vote for leaving as-is; subclasses that extend `ConcurrentResourceCache` can always add in class-specific logs for clarity if needed.

Reply:
I agree. I ended up adding this to the constructor of `ConcurrentResourceCache`:
```java
logger.info("Initialized cache {} with ID {}", this.getClass().getName(), this.cacheID);
```

It will print the subclass name if this constructor was being invoked thru a `super`. Should give @jklamer the search info he needs without adding more noise to every log message.